### PR TITLE
feat(defaultOption): implements defaultOption

### LINKED ISSUE #200: defaultOption for a type
http://angular-formly.com/#/example/custom-types/default-options

### PR DESCRIPTION
**Pull request by divyakumarjain**
implements defaultOption which helps in defining reusable types and keeping configuration short

Closes #200

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ node_modules
 npm-debug.log
 yarn-error.log
 coverage
+typings
```
Comment (aitboudad) [inline .gitignore]: We don't use typings anymore, instead can you remove typings folder in your local

File: package.json [hunk 2]
```diff
@@ -22,9 +22,9 @@
     "build:lib": "npm run ngc -- --project tsconfig.build.json",
     "prebuild:bundles": "shx rm -rf ./bundles",
     "build:bundles": "npm run build:bundles:compile && npm run build:bundles:minify",
-    "build:bundles:compile": "node ./node_modules/.bin/webpack --config .config/webpack.build.js",
+    "build:bundles:compile": "webpack --config .config/webpack.build.js",
     "build:bundles:minify": "uglifyjs bundles/ng2-formly.umd.js --output bundles/ng2-formly.umd.min.js --in-source-map bundles/ng2-formly.umd.js.map --source-map bundles/ng2-formly.umd.min.js.map",
-    "demo": "NODE_ENV=dev node node_modules/.bin/webpack-dashboard --title ng2-formly-demo -- node_modules/.bin/webpack-dev-server --quiet --port 9000 --content-base demo --config .config/webpack.demo.js --open",
+    "demo": "webpack-dashboard --title ng2-formly-demo -- node_modules/.bin/webpack-dev-server --quiet --port 9000 --content-base demo --config .config/webpack.demo.js --open",
```
Comment (aitboudad) [inline package.json]: This will require to install webpack-dashboard globally , I prefer the old one

File: src/core/components/formly.field.ts [hunk 3]
```diff
@@ -57,9 +57,23 @@ export class FormlyField implements DoCheck, OnInit {
   }
 
   ngOnInit() {
+    this.initDefaultOptions();
```
Comment (aitboudad) [inline src/core/components/formly.field.ts]: I would prefer to move this into formly-form component

File: demo/hello.ts [hunk 4]
```diff
@@ -461,6 +451,22 @@ export class HelloApp {
         { name: 'formly-wrapper-horizontal', component: FormlyWrapperHorizontalLabel, types: ['horizontalInput'] },
         { name: 'panel', component: FormlyPanelWrapper },
       ],
+      defaultOptions: [
+        {
```
Comment (mohammedzamakhan) [inline demo/hello.ts]: defaultOptions is not meant to be set separately, its set as a part of the setType.

File: package.json (hunk 2 above)
Comment (divyakumarjain) [inline package.json]: Reverting it but i was not able to run old one one windows machine.

File: demo/hello.ts (hunk 4 above)
Comment (divyakumarjain) [inline demo/hello.ts]: @mohammedzamakhan after looking at PR #232 seems like the direction in this PR is different then intended. Let me know if you want to me to close this PR and you will continue with #232 .

File: package.json (hunk 2 above)
Comment (divyakumarjain) [inline package.json]: raised new issue https://github.com/formly-js/ng2-formly/issues/233 for this

**Pull request closed without merging**